### DISCUSSION TITLE
Should we remove the option to include an .eps file?

### DISCUSSION THREAD

**Discussion by Snailedlt** [devicons/devicon#1302] [Ideas]
I'm not sure exactly what the benefit of an .eps file is, but I can come up with a few disadvantages:
1. **More work than leaving it out:** Pretty self explanatory, but it's more work making something, than not making something
2. **More code to maintain:** If you want to update an icon, you'll also need to update the following .eps file
3. **Can't be viewed on GitHub | Harder to review:** Which means you'll need to download the branch/PR to make sure the .eps file is okey. Bye-bye reviews on the phone 😢 
4. **Higher bar of entry:** Not everyone (me included) knows exactly what an .eps file is, nor how to make it. In short, more complexity, and higher bar of entry for contributors.

Like I said, I don't know much about .eps file, or the benefit of having them, but if you do, please share

**Comment by Snailedlt** [comment]
@BenSouchet I've seen you add .eps files a lot to your PR's, perhaps you could give some input here?

**Comment by BenSouchet** [comment]
@Snailedlt

EPS file are cross software vector file.
These files was used, and are still used in some situations, in multiples domaines.

EPS are still used, for example, on the digital / media industry, if you need to send a vectorised image / logo / sign to a printing company, they will ask you an EPS file not an `.ai`, `.svg`, `.psd`, ...

### Is `.eps` files useful for Devicon ?
For devicon the answer is a bit hard.
On one side, since we can open, edit, re-save SVGs easily with almost every image / vector processing softwares EPS seems to be useless.

**But** sometime, SVGs are so "optimized" that the paths are merged / simplified, so you cannot easily re-edit the vectorised logo / icon.
If you have the source EPS file you could do the modifications, since normally EPS contains the vector elements (paths) not optimized, so it's fully editable.

**Now my answer,** I think EPS files isn't needed for Devicon since after being added icons won't be edited, maybe someone will update the icon (if a brand has changed its logo) but this person will not edit the current file (EPS file or SVGs).

### Why I add EPS files in my PRs ?

When I contributed for the first time I looked whathas been done by others, there was often a EPS file created, so I did the same thing.
And since then I've continued to add it :)

**Reply by BenSouchet** [reply]
One other element to take into account is that in addition to taking extra time to create these files are often quite heavy, so if no one use them we could save a lot of disk space on the repo, and locally when we `git pull` the project!

**Reply by Snailedlt** [reply]
Wow, thank you for this amazingly detailed response! I learned a lot, and feel much better equipped to make an informed decision now.

I see both the negative and positive sides here now that you explain it. I tend to agree with you on the conclusion, and I do here too. I still want to make sure it's okay for the other maintainers that this is removed from the wiki and contributing guidelines before I make any changes though.

**Reply by Snailedlt** [reply]
@devicons/maintainer Please read the message by @BenSouchet, and leave a comment on your opinion on this.
I personally agree with ben that the .eps files should not be recommended in the wiki, and perhaps even the existing ones removed.

**Reply by Panquesito7** [reply]
I believe we should add this as optional and keep the current ones. If an icon is being updated and no `.eps` file is provided, we can remove it or update it (this would be the best option, IMO).

If someone wants to add it to their PR, that's great, then; there might be someone that uses this format for their work or projects.
In short: make the `.eps` addition optional, keep the current ones, and remove it if an icon is being updated.

What do you think about this?

**Reply by BenSouchet** [reply]
@Snailedlt 
For the removal of existing EPS files I think this will require a new major release of Devicon, because maybe someone somewhere rely on existing EPS files.

On the topic of a new major release of Devicon I posted [a comment](https://github.com/devicons/devicon/discussions/353#discussioncomment-1981972) talking about some element that could be done in this new major release.
Maybe we could / should also create a discussion to list element that could be done on the next major version (just an idea).

**Reply by Snailedlt** [reply]
@Panquesito7 
.eps are already optional though.

*(sorry in advance if I'm being a bit brute)*
I see your point, but I disagree with the conclusion, mainly because of the following reasons:
1. (main reason) .eps files increase review time by a lot. This is because you have to checkout the PR locally in order to view the file, since it can't be viewed on GitHub.
2. like BenSouchet mentioned .eps add to complexity and take up a lot of space
3. I feel .eps files fall outside of the natural scope of Devicons as a whole. The reason being that devicons is made for devs to easily implement dev related icons into their projects. .eps files don't really help with that as far as I can tell.

**Reply by Snailedlt** [reply]
> @Snailedlt For the removal of existing EPS files I think this will require a new major release of Devicon, because maybe someone somewhere rely on existing EPS files.
> 
> On the topic of a new major release of Devicon I posted [a comment](https://github.com/devicons/devicon/discussions/353#discussioncomment-1981972) talking about some element that could be done in this new major release. Maybe we could / should also create a discussion to list element that could be done on the next major version (just an idea).

@BenSouchet Yes I agree. Would you be willing to make an issue or a discussion about this?

**Reply by BenSouchet** [reply]
> @BenSouchet Yes I agree. Would you be willing to make an issue or a discussion about this?

@Snailedlt Not have much time right now but yes will create a discussion soon.
